src/main/java/com/mraof/minestuck/alchemy/PredeterminedCardCaptchas.java

```diff
@@ -0,0 +1,132 @@
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.JsonOps;
+import com.mraof.minestuck.Minestuck;
+import net.minecraft.MethodsReturnNonnullByDefault;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.packs.resources.ResourceManager;
+import net.minecraft.server.packs.resources.SimplePreparableReloadListener;
+import net.minecraft.util.profiling.ProfilerFiller;
+import net.minecraft.world.item.Item;
+import net.minecraftforge.event.AddReloadListenerEvent;
+import net.minecraftforge.event.server.ServerStoppedEvent;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.registries.ForgeRegistries;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import javax.annotation.Nullable;
+import javax.annotation.ParametersAreNonnullByDefault;
+import java.io.IOException;
+import java.io.Reader;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+@ParametersAreNonnullByDefault
+@MethodsReturnNonnullByDefault
+@Mod.EventBusSubscriber(modid = Minestuck.MOD_ID, bus = Mod.EventBusSubscriber.Bus.FORGE)
+public class PredeterminedCardCaptchas
+{
+	private static BiMap<Item, String> predefinedCardMap = HashBiMap.create();
+	
+	private static void setData(BiMap<Item, String> predefinedCards)
+	{
+		PredeterminedCardCaptchas.predefinedCardMap = predefinedCards;
+	}
+	
+	public static BiMap<Item, String> getData()
+	{
+		return Objects.requireNonNull(PredeterminedCardCaptchas.predefinedCardMap, "Tried to get an instance of Predetermined Captchas too early.");
+	}
+	
+	@SubscribeEvent
+	public static void onServerStopped(ServerStoppedEvent event)
+	{
+		predefinedCardMap = null;
+	}
+	private final static class Loader extends SimplePreparableReloadListener<Map<String, Item>>
+	{
+		private static final Logger LOGGER = LogManager.getLogger();
+		public static final String PATH = "minestuck/captcha_codes.json";
+		@Override
+		protected Map<String, Item> prepare(ResourceManager resourceManager, ProfilerFiller profiler)
+		{
+			BiMap<String, Item> captchaData = HashBiMap.create();
+			
+			for(String namespace : resourceManager.getNamespaces())
+			{
+				parseCaptchaCodesFromNamespace(resourceManager, namespace).ifPresent(parsedData -> processCaptchaData(captchaData, parsedData));
+			}
+			return captchaData;
+		}
+		
+		@Override
+		protected void apply(Map<String, Item> data, ResourceManager resourceManager, ProfilerFiller profiler)
+		{
+			ImmutableBiMap.Builder<Item, String> predefinedCards = ImmutableBiMap.builder();
+			
+			for(Map.Entry<String, Item> entrySet : data.entrySet())
+			{
+				predefinedCards.put(entrySet.getValue(), entrySet.getKey());
+			}
+			
+			PredeterminedCardCaptchas.setData(predefinedCards.build());
+		}
+		
+		@SubscribeEvent
+		public static void onResourceReload(AddReloadListenerEvent event)
+		{
+			event.addListener(new Loader());
+		}
```

Comment:
This doesn't seem to be working since it's not directly inside the class with the `@EventBusSubscriber` annotation. A simple solution is to move it up to `PredeterminedCardCaptchas`.